src/west/commands/project.py

```diff
@@ -44,13 +44,16 @@ def do_add_parser(self, parser_adder):
                            _arg('-a', '--abspath', action='store_true',
                                 help='''just print absolute paths (if given
                                 with -p, this option takes precedence)'''),
+                           _arg('-s', '--special', action='store_true',
+                                help='''include output for "special" projects
+                                (west itself and the manifest repository)'''),
```

Comment:
I think it would be more general to have `--projects`, `--west`, and `--manifest` flags. Then you could just list `west` if you wanted to, for example, without getting the other projects in there.

To list just the absolute path to `west`, you'd do

    west list --west --format="{abspath}"

The default would be something like

    west list --projects --west --manifest --format="<default format string>"

Reply:
I agree that @ulfalizer's proposal feels cleaner. I would have the following:

* `west list` by default lists `--projects`
* `west list -p/--projects`
* `west list -w/--west`
* `west list -m/--manifest`
* `west list -a/--all`

I don't really think we need the `--format` instead of the `--abspath` but I don't mind

Reply:
`--format` could be a cleaner way, especially if that is a general way to format output from `west`.
That way, user doesn't have to think, should I use `-p` or `-a` or `--something-else` in the various `west <command>` options.

There will always be a `--format`, although what and  how to format may vary.

@carlescufi `--porcelain` is normally intended for humans and have higher risk of changing formatting. Whereas for a script I guess a low-level output command with fixed output is more suited.

Reply:
@tejlmand from `git help status`:

```
       --porcelain[=<version>]
           Give the output in an easy-to-parse format for scripts. This is similar to the short output, but will remain stable across Git versions and regardless of user configuration. See below for details.

           The version parameter is used to specify the format version. This is optional and defaults to the original version v1 format.
```

Reply:
Or actually to simplify another option would be to have:

`west list` (lists all projects)
and
`west list --all` (lists all projects + west + manifest)

Reply:
Personally, I would get rid of `--all` and have `west list` act as an alias for `west list --west --manifest --projects`.

That's easy to explain and understand, and everything else can be implemented in terms of it.

Similarly, leaving out `--format` would just substitute some default hardcoded `--format` string in the code, so that everything can go through the same path.

I'm not sure we need some kind of `--porcelain` command if we have the `--format` parameter, at least not yet. If something turns out to be tricky to parse, we could always add more replacement fields that do porcelainish stuff.

Reply:
`--format` is better than having extra flags for particular hardcoded formats I think, especially if we need to add more stuff in the future.

I think it's a good idea to see if one thing can be implemented in terms of another thing, and to try to go for the most "powerful" construct in those terms.

Reply:
> 
> Personally, I would get rid of `--all` and have `west list` act as an alias for `west list --west --manifest --projects`.
> 

Not so fond of having specific projects directly in the arguments, such as: `--manifest`, `--west`.
I would prefer those to be values to an argument, such as `--special west` or `--projects west`, depending on what we settle upon.

> Similarly, leaving out `--format` would just substitute some default hardcoded `--format` string in the code, so that everything can go through the same path.
> 
> I'm not sure we need some kind of `--porcelain` flag if we have the `--format` flag, at least not yet. If something turns out to be tricky to parse, we could always add more replacement fields that do porcelainish stuff.

Currently, I think the `--format` is the nicest solution, but we should ensure good description / examples in the help text, so people don't have to look in source code to find out values to print.

`--porcelain` could be thought of as just one particular fixed formatting string, if we would like to make easy to use, for majority.